File: core/src/com/g3ida/withflyingcolours/core/player/controller/PlayerControllerSystem.java
package com.g3ida.withflyingcolours.core.player.controller;

import com.artemis.ComponentMapper;
import com.artemis.annotations.All;
import com.artemis.systems.IteratingSystem;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Input;

import games.rednblack.editor.renderer.utils.ComponentRetriever;

@All(PlayerControllerComponent.class)
public class PlayerControllerSystem extends IteratingSystem {
    ComponentMapper<PlayerControllerComponent> mPlayerControllerComponent;

    public PlayerControllerSystem() {
        super();
    }

    @Override
    protected void process(int entityId) {
        //player movement left/right
        PlayerControllerComponent playerMovement = mPlayerControllerComponent.get(entityId);
        if (Gdx.input.isKeyPressed(Input.Keys.LEFT)) {
            playerMovement.moveInput = -1;
        } else if (Gdx.input.isKeyPressed(Input.Keys.RIGHT)) {
            playerMovement.moveInput = 1;
        } else {
            playerMovement.moveInput = 0;
        }
        //jumping
        playerMovement.shouldJump = Gdx.input.isKeyPressed(Input.Keys.UP);
        //rotating
        playerMovement.shouldRotateLeft = Gdx.input.isKeyJustPressed(Input.Keys.Z);
        playerMovement.shouldRotateRight = Gdx.input.isKeyJustPressed(Input.Keys.C);
    }
}

File: core/src/com/g3ida/withflyingcolours/core/player/controller/PlayerControllerSystem.kt
package com.g3ida.withflyingcolours.core.player.controller

import com.artemis.ComponentMapper
import com.artemis.systems.IteratingSystem
import com.artemis.annotations.All
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.Input

@All(PlayerControllerComponent::class)
class PlayerControllerSystem : IteratingSystem() {
    lateinit var mPlayerControllerComponent: ComponentMapper<PlayerControllerComponent>
    override fun process(entityId: Int) {
        //player movement left/right
        val playerMovement = mPlayerControllerComponent[entityId]
        playerMovement.moveInput = when {
            Gdx.input.isKeyPressed(Input.Keys.LEFT) -> -1
            Gdx.input.isKeyPressed(Input.Keys.RIGHT) -> 1
            else -> 0
        }
        //jumping
        playerMovement.shouldJump = Gdx.input.isKeyPressed(Input.Keys.UP)
        //rotating
        playerMovement.shouldRotateLeft = Gdx.input.isKeyJustPressed(Input.Keys.Z)
        playerMovement.shouldRotateRight = Gdx.input.isKeyJustPressed(Input.Keys.C)
    }
}